stm32: cmsis: spurious build issues

#### Description
I notice spurious build issues when building an stm32 based board on the CI logs for quite some time now.  The issue is with the cmsis vendor headers, which are cloned during the build process. 

An example log is here: https://ci.riot-os.org/RIOT-OS/RIOT/15337/bf9f5cce1f0c77cab947bb3c37d35516c5ed3733/output/compile/tests/riotboot/nucleo-l552ze-q:gnu.txt


#### Steps to reproduce the issue
I can also reproduce this problem locally on my machine with the following command:
```
rm -rf cpu/stm32/include/vendor/cmsis/
BOARD=nucleo-l552ze-q make -j4 -C tests/riotboot clean all
```
The issue only manifests if build with multiple processes (which is done on the CI all the time).

#### Expected results
Compilation succeeds without warnings or errors.

#### Actual results
I oftentimes see a git problem like:
```
make: Entering directory 'RIOT/tests/riotboot'
Building application "tests_riotboot" for "nucleo-l552ze-q" with MCU "stm32".

[INFO] cloning stm32cmsis
git-cache: cloning from cache. tag=commitd922865fc0326a102c26211c44b8e42f52c1e53d-33802
compiling RIOT/dist/tools/riotboot_gen_hdr/bin/genhdr...
make: Nothing to be done for 'all'.
[INFO] updating stm32cmsis RIOT/cpu/stm32/include/vendor/cmsis/l5/.pkg-state.git-downloaded
[INFO] updating stm32cmsis RIOT/cpu/stm32/include/vendor/cmsis/l5/.pkg-state.git-downloaded
echo d922865fc0326a102c26211c44b8e42f52c1e53d   > RIOT/cpu/stm32/include/vendor/cmsis/l5/.pkg-state.git-downloaded
[INFO] patch stm32cmsis
echo d922865fc0326a102c26211c44b8e42f52c1e53d   > RIOT/cpu/stm32/include/vendor/cmsis/l5/.pkg-state.git-downloaded
[INFO] patch stm32cmsis
fatal: Unable to create 'RIOT/cpu/stm32/include/vendor/cmsis/l5/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
make[1]: *** [RIOT/pkg/pkg.mk:110: RIOT/cpu/stm32/include/vendor/cmsis/l5/.pkg-state.git-patched] Error 128
make: *** [RIOT/cpu/stm32/Makefile.include:66: RIOT/cpu/stm32/include/vendor/cmsis/l5/Include/stm32l5xx.h] Error 2
make: *** Waiting for unfinished jobs....
```

#### Versions
```

Operating System Environment
----------------------------
         Operating System: "Arch Linux" 
                   Kernel: Linux 5.9.6-arch1-1 x86_64 unknown
             System shell: GNU bash, version 5.0.18(1)-release (x86_64-pc-linux-gnu)
             make's shell: GNU bash, version 5.0.18(1)-release (x86_64-pc-linux-gnu)

Installed compiler toolchains
-----------------------------
               native gcc: gcc (GCC) 10.2.0
        arm-none-eabi-gcc: arm-none-eabi-gcc (Arch Repository) 10.2.0
                  avr-gcc: avr-gcc (GCC) 10.2.0
         mips-mti-elf-gcc: missing
           msp430-elf-gcc: missing
       riscv-none-elf-gcc: missing
  riscv64-unknown-elf-gcc: missing
     riscv-none-embed-gcc: missing
     xtensa-esp32-elf-gcc: missing
   xtensa-esp8266-elf-gcc: missing
                    clang: clang version 11.0.0

Installed compiler libs
-----------------------
     arm-none-eabi-newlib: "3.3.0"
      mips-mti-elf-newlib: missing
        msp430-elf-newlib: missing
    riscv-none-elf-newlib: missing
riscv64-unknown-elf-newlib: missing
  riscv-none-embed-newlib: missing
  xtensa-esp32-elf-newlib: missing
xtensa-esp8266-elf-newlib: missing
                 avr-libc: "2.0.0" ("20150208")

Installed development tools
---------------------------
                   ccache: ccache version 3.7.11
                    cmake: cmake version 3.18.4
                 cppcheck: Cppcheck 2.2
                  doxygen: 1.8.20
                      git: git version 2.29.2
                     make: GNU Make 4.3
                  openocd: Open On-Chip Debugger 0.10.0
                   python: Python 3.8.6
                  python2: Python 2.7.18
                  python3: Python 3.8.6
                   flake8: error: /usr/bin/python3: No module named flake8
               coccinelle: missing
```

--- a/makefiles/boot/riotboot.mk
+++ b/makefiles/boot/riotboot.mk
@@ -76,7 +76,7 @@ riotboot: $(SLOT_RIOT_BINS)
 
 # riotboot bootloader compile target
 riotboot/flash-bootloader: riotboot/bootloader/flash
-riotboot/bootloader/%:
+riotboot/bootloader/%: $(BUILDDEPS)
 	$(Q)/usr/bin/env -i \
 		QUIET=$(QUIET) PATH="$(PATH)"\
 		EXTERNAL_BOARD_DIRS="$(EXTERNAL_BOARD_DIRS)" BOARD=$(BOARD)\

